src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxr.cpp

```diff
@@ -87,7 +88,14 @@ void HostFxr::SetMain(hostfxr_main_fn hostfxr_main_fn)
 
 int HostFxr::Main(DWORD argc, const PCWSTR* argv) const noexcept(false)
 {
-    return m_hostfxr_main_fn(argc, argv);
+    if (m_host_context_handle != nullptr && m_hostfxr_run_app_fn != nullptr)
+    {
+        return m_hostfxr_run_app_fn(m_host_context_handle);
+    }
+    else
+    {
+        return m_hostfxr_main_fn(argc, argv);
```

Comment:
We're losing the environment variables that were set before, is that ok?

Reply:
Yeah, it should be okay. All of this code lives in the handler, which is part of the aspnetcore runtime. Therefore, the only reason this function would be called is from our testing framework, where we set a static main method to be called here. Because everything lives in the handler, the handler version and the hostfxr version should align, meaning the handler will have access to all hostfxr functions necessary.